samples/test-app-without-backward-compatibility/src/androidTest/kotlin/com/avito/android/test/app/second/SnackbarProxyTest.kt

```diff
@@ -0,0 +1,111 @@
+package com.avito.android.test.app.second
+
+import com.avito.android.test.app.core.screenRule
+import com.avito.android.ui.test.Screen
+import org.hamcrest.Matchers
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+
+class SnackbarProxyTest {
+
+    @get:Rule
+    val rule = screenRule<SnackbarProxyTestActivity>()
+
+    @get:Rule
+    val snackbarRule = SnackbarRule()
+
+    val snackbarChecks = snackbarRule.checks
+
+    @get:Rule
+    val exception: ExpectedException = ExpectedException.none()
+
+    @Test
+    fun check_snackbar_is_shown() {
+        rule.launchActivity(null)
+
+        Screen.snackbarProxyScreen.clickShowSnackbar()
+
+        snackbarChecks.isShownWithExactlyText("snackbar number 1")
+    }
+
+    @Test
+    fun check_snackbar_is_shown_last() {
+        rule.launchActivity(null)
+
+        Screen.snackbarProxyScreen.clickShowSnackbar()
+        Screen.snackbarProxyScreen.clickShowSnackbar()
+
+        snackbarChecks.isShownLastWithExactlyText("snackbar number 2")
+    }
+
+    @Test
+    fun check_snackbar_is_shown_N_times() {
```

Comment:
Why do we need to support this check? Have we seen it in tests?

Reply:
Asked users. As for me, it looks preferable

Reply:
Scenarios
- Check zero times when you don't want a snackbar
- Check two times because you have an old one snackbar

Reply:
Ok, zero times looks like a valid case.